core/execution/amm/engine.go

```diff
@@ -157,6 +158,7 @@ func New(
 	marketActivityTracker *common.MarketActivityTracker,
 	parties common.Parties,
 ) *Engine {
+	oneTick, _ := num.UintFromDecimal(num.DecimalOne().Mul(priceFactor))
```

Comment:
I don't understand a couple of things about this:
1. this will always be 0 if the asset has less decimals than the market, is that ok?
2. what's the point of multiplying by 1?
3. why should it be price factor rather than just 1 if it's prices in asset than adjusting by 1 is what you want isn't it?

Reply:
1) probably not, but none of this has been tested with negative DP's yet since they appeared half-way through the AMM feature branch's life and I just had to resolve the conflicts. Its been on my list to revist for a while, I'll make a ticket for it now.
2) there is no point, I'll fix it up
3) the matching algorithm for AMMs as defined in the spec works to the resolution of order-price-levels, there is no need for it to be more granual because any incoming orders will only be at these price-factor price levels and so knowing that an AMM's fair-price is "somewhere between two price levels" is enough. Whenever we need to view an AMM as orders, or if we need to reason about what a "unit-step" is along the AMM's curve we only need to consider the order-price levels to ensure everything remains uncrossed.

For example here in this bug, when we need to uncross at auction end, we "expand" the AMM in orders to this one-tick granuality because thats all we need to ensure we remain uncrossed afterwards. Then when we submit those orders to uncross, we again only need to reason in this one-tick granuality. 

We could instead view "unit-steps" along the AMM's curve in asset prices but this would not only cause a performance hit because there would be so many. We would also need to enforce that the commitment of an AMM is sufficient to create at least 1 volume between these "unit-steps" on the AMM curve so that the maths doesn't break down with `< 1` volume price changes. If we use asset-prices for our "unit-steps" then the minimum commitment for an AMM would have to me much much higher so that this remains true with smaller steps.